### PR TITLE
feat(@core/home) : add home 

### PR DESCRIPTION
**Pull request by marobakr**
1. add home

### CODE REVIEW

**Comment by vcpc** [comment]
1. padding top in the small screen version is only 82px not 132px:
image
---
2. the photo and the description is smaller than the design
Your Work:
image
The Design:
image

**Comment by marobakr** [comment]
1. add home
1.1 smaller padding-top
1.2 bigger font size in the description
1.3 I didn't change the width of the photos because in small media at Figma, it was width: 313px and I designed it like Figma

**Comment by marobakr** [comment]
1. home
1.1 smaller padding-top
1.2 center image in small media
1.3. handle comments in style Helper

2. button
2.1. add global rule  btn-main-primary for all buttons
2.2. handle button on hover 
2.3. add global rule  btn-main-primary for all buttons
2.4. remove duplicate roule in variables at line number 23
2.5. change role name from  primaryBorder to border-primary 
2.6. change role name from  grayBorder to border-secondary

**Comment by marobakr** [comment]
1. home
2. button

1.1 smaller padding-top
1.2 center image in small media
1.3. handle comments in style Helper

2.1. add global rule btn-main-primary for all buttons
2.2. handle button on hover
2.3. add global rule btn-main-primary for all buttons
2.4. remove duplicate roule in variables at line number 23
2.5. change role name from primaryBorder to border-primary
2.6. change role name from grayBorder to border-secondary

If applied, "it will be change design  an image and button contact in the home "

File: src/app/styles/components/_button.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
+button.btn-main-primary {
+  /* Box Model */
+  padding: 8px 16px;
+  border: 1px solid #c778dd;
```
Comment (vcpc) [inline src/app/styles/components/_button.scss:4]: import the `_variables.scss` file and use the `$primary` variable

File: src/app/styles/utilities/_variables.scss [hunk 2]
```diff
@@ -19,7 +15,25 @@ $font: "Fira Code";
   /* Color */
   background-color: $gray;
 }
+.bg-main-primary {
+  /* Color */
+  background-color: $primary;
+}
+
+.text-main {
```
Comment (vcpc) [inline src/app/styles/utilities/_variables.scss:23]: why did you remove the `color-main`, WE agreed that the one that will be removed is the `text-main`
this canceled all the main colored elements in the page !

**Comment by vcpc** [comment]
1. the photo is not centered in the small screen version
2. use the [BreakpointObserver](https://blog.angular-university.io/angular-responsive-design/) for the responsive design, if you don't know to use it ask Eng. Mohamed H.

**Comment by marobakr** [comment]
1. home
1.1. import _variables.scss  file and use  $primary 
1.2. add text-white in  _button.scss file
1.3. replace rule the text-main with color-main in  _variables.scss 
1.4 center image in small media

**Comment by marobakr** [comment]
1.home
1.1. import _variables.scss file and use $primary
1.2. add text-white in _button.scss file
1.3. replace rule the text-main with color-main in _variables.scss
1.4 center image in small media

**Comment by marobakr** [comment]
1.handle small media with observer

File: src/app/styles/components/_button.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  border: 1px solid $primary;
+  border-radius: 0;
+  color: white;
+
+  /* Other */
+  transition: all 300ms ease-out;
+  &:hover {
+    /* Color */
+    background-color: rgba($color: #c778dd, $alpha: 20%);
+    border: 1px solid #c778dd;
```
Comment (vcpc) [inline src/app/styles/components/_button.scss:14]: this line is useless

File: src/app/core/components/home/home.component.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
+<section [ngClass]="{ 'small-media': mobileMedia }" class="home">
+  <div class="container px-3">
+    <div
+      class="row gx-2 align-items-center justify-content-between home__row-mb"
```
Comment (vcpc) [inline src/app/core/components/home/home.component.html:4]: this property string is more than 60 char
what is the use of this class `.home__row-mb` ?

File: src/app/core/components/home/home.component.scss [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  &.small-media {
+    padding-top: 82px; /* Box Model */ /* 2 */
+  }
+
+  /* Box Model */
+  padding-top: 123px;
+  min-height: 100vh;
+
+  /* Color */
+  background-color: #282c33;
```
Comment (vcpc) [inline src/app/core/components/home/home.component.scss:17]: use the `$background` variable

File: src/app/styles/utilities/_variables.scss [hunk 6]
```diff
@@ -19,7 +15,25 @@ $font: "Fira Code";
   /* Color */
   background-color: $gray;
 }
+.bg-main-primary {
+  /* Color */
+  background-color: $primary;
+}
+
+.color-main {
```
Comment (vcpc) [inline src/app/styles/utilities/_variables.scss:23]: change this to `text-main`, this is making trouble in the navbar styling

**Comment by vcpc** [comment]
1. you reformatted the `navbar.component,html` file, copy it's original content from the repo to undo your changes
2. the photo is not centered
image
---
3. fix conflicts
